src/geometry/manifold/integral_curve.lean

```diff
@@ -0,0 +1,287 @@
+    apply set.mem_of_mem_of_subset _
+      (interior_subset : interior 𝓔(I, x₀).target ⊆ 𝓔(I, x₀).target),
+    rw ←set.mem_preimage,
+    exact hf3 },
+  { rw [function.comp_apply, ←set.mem_preimage, ←set.mem_preimage],
+    apply set.mem_of_mem_of_subset _ (set.inter_subset_right 𝓔(I, x₀).target _),
+    rw [local_equiv.target_inter_inv_preimage_preimage,
+      set.inter_eq_self_of_subset_right interior_subset],
+    exact hf3 },
+  { refine continuous_at.comp _ hf4.continuous_at,
+    apply ext_chart_continuous_at_symm'',
+    exact set.mem_of_mem_of_subset hf3 interior_subset },
+  { rw [function.comp_apply, ←function.comp_apply v,
+    ←function.comp_apply 𝓔(I.tangent, v x₀), ←written_in_ext_chart_at],
+    apply has_deriv_at.congr_of_eventually_eq hf4,
+    rw filter.eventually_eq_iff_exists_mem,
+    refine ⟨metric.ball 0 ε, is_open.mem_nhds metric.is_open_ball ht, _⟩,
+    intros t' ht',
+    rw [function.comp_apply, function.comp_apply],
+    apply local_equiv.right_inv,
+    exact set.mem_of_mem_of_subset (hf2 t' ht').1 interior_subset }
+end
+
+-- how to generalise / simplify?
+/-- The derivative of a curve on a manifold is independent of the chosen extended chart. -/
+lemma curve_has_deriv_at_coord_change
+  (v : M → tangent_bundle I M) (h₁ : ∀ x, (v x).1 = x) (x₀ : M) (γ : ℝ → M) (t : ℝ)
+  (hγ₁ : (γ t) ∈ 𝓔(I, x₀).source)
+  (hγ₂ : 𝓔(I, x₀) (γ t) ∈ interior 𝓔(I, x₀).target)
+  (hd : has_deriv_at (𝓔(I, x₀) ∘ γ) (𝓔(I.tangent, v x₀) (v (γ t))).2 t) :
+  has_deriv_at ((𝓔(I, γ t) ∘ 𝓔(I, x₀).symm) ∘ (𝓔(I, x₀) ∘ γ))
+    (𝓔(I.tangent, v (γ t)) (v (γ t))).2 t :=
+begin
+  have : (v (γ t)).fst ∈ 𝓔(I, (v x₀).1).source,
+  { rw [h₁, h₁],
+    exact hγ₁ },
+  rw tangent_bundle_core_coord_change_triv' I M (v (γ t)) (v x₀) this,
+  apply has_fderiv_at.comp_has_deriv_at _ _ hd,
+  rw [h₁, h₁, function.comp_apply],
+  have : set.range I ∈ nhds (𝓔(I, x₀) (γ t)),
+  { rw mem_nhds_iff,
+    refine ⟨interior 𝓔(I, x₀).target, _, is_open_interior, hγ₂⟩,
+    refine set.subset.trans interior_subset _,
+    rw ext_chart_at_target,
+    exact set.inter_subset_right _ _ },
+  apply has_fderiv_within_at.has_fderiv_at _ this,
+  apply differentiable_within_at.has_fderiv_within_at,
+  apply cont_diff_within_at.differentiable_within_at _ le_top,
+  apply cont_diff_within_at_ext_coord_change,
+  apply local_equiv.mem_symm_trans_source _ hγ₁,
+  exact mem_ext_chart_source _ _
+end
+
+/-- For any continuously differentiable vector field and any chosen non-boundary point `x₀` on the
+  manifold, an integral curve `γ : ℝ → M` exists such that `γ 0 = x₀` and the tangent vector of `γ`
+  at `t` coincides with the vector field at `γ t` for all `t` within an open interval around 0.-/
+theorem exists_integral_curve_of_cont_mdiff_tangent_vector_field [proper_space E]
+  (v : M → tangent_bundle I M) (h₁ : ∀ x, (v x).1 = x) (h₂ : cont_mdiff I I.tangent 1 v)
+  (x₀ : M) (hx : 𝓔(I, x₀) x₀ ∈ interior 𝓔(I, x₀).target) :
```

Comment:
Instead of `hx`, I'd rather define a predicate saying that `x₀` is a non-boundary point, and use that here and in the other statements above.

Reply:
Implemented! I also added `t₀` as an argument so it is not assumed to be =0.